service/create/service.go

```diff
@@ -54,37 +55,45 @@ const (
 // Config represents the configuration used to create a version service.
 type Config struct {
 	// Dependencies.
-	AwsConfig  awsutil.Config
-	K8sClient  kubernetes.Interface
-	Logger     micrologger.Logger
-	S3Bucket   string
-	PubKeyFile string
+	// TODO this is a setting, not a dependency.
+	AwsConfig   awsutil.Config
+	CertWatcher *certkit.Service
+	K8sClient   kubernetes.Interface
+	Logger      micrologger.Logger
+	S3Bucket    string
+	PubKeyFile  string
 }
 
 // DefaultConfig provides a default configuration to create a new service by
 // best effort.
 func DefaultConfig() Config {
 	return Config{
 		// Dependencies.
-		K8sClient:  nil,
-		Logger:     nil,
-		S3Bucket:   "",
-		PubKeyFile: "",
+		CertWatcher: nil,
+		K8sClient:   nil,
+		Logger:      nil,
+		S3Bucket:    "",
+		PubKeyFile:  "",
 	}
 }
 
 // New creates a new configured service.
 func New(config Config) (*Service, error) {
 	// Dependencies.
+	// TODO other deps have to be checked as well
```

Comment:
The config is divided in dependency and settings sections. A dependency is something that implements business logic. A setting is basically data. Logger and k8s client are dependencies required to make the service actually work. Therefore the constructor here should make sure that sufficient configuration is provided. At the moment I can just forget to inject a k8s client and the operator will fail on startup or panic during runtime.